### PR TITLE
Add preselected cipher suites to dtls configuration.

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 1]
```diff
@@ -3413,7 +3512,16 @@ private void determineCipherSuitesFromConfig() {
...
-
+			if (config.preselectedCipherSuites != null) {
+				// remove not preselected cipher suites
+				Iterator<CipherSuite> iterator = ciphers.iterator();
+				while (iterator.hasNext()) {
+					CipherSuite cipherSuite = iterator.next();
+					if (!config.preselectedCipherSuites.contains(cipherSuite)) {
+						iterator.remove();
+					}
+				}
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java:3523]: It seems to me that order will be defined by default cipher suite order.
Should we rather use the `preselectedCipherSuites` cipher suite order ?

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 2]
```diff
@@ -313,6 +314,20 @@
 	 */
 	private CipherSuiteSelector cipherSuiteSelector;
 
+	/**
+	 * Preselected cipher suites.
+	 * 
+	 * If no supported cipher suites are provided, consider only this subset of
+	 * {@link CipherSuite} to be selected as supported cipher suites for this
+	 * peer.
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java:322]:
```suggestion
	 * If no supported cipher suites are provided, consider only this subset of
	 * {@link CipherSuite} to be automatically selected as supported cipher suites depending 
	 * on other setting (e.g. if settings allow only PSK, only PSK compatible cipher suite from this list will be selected). 
```

Maybe this deserve a bit more explanation.
I propose something but feel free to adapt or just keep it like this. :slightly_smiling_face:

**Review by sbernard31 — COMMENTED** [review]
Tested and it works for me.

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java (hunk 1 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java:3523]: The order only complies to sub-sections in cipher suites. But the order of theses sections is:

- ECDSA
- ECDHE_PSK
- PSK

Anyway, sure we can use the order of the preselected cipher suites.

I adapt it.

**Comment by boaks** [comment]
I adapted the order and the javadocs.

**Review by sbernard31 — APPROVED** [review]
LGTM.

Thx @boaks for this new feature :pray:

**Pull request merged**